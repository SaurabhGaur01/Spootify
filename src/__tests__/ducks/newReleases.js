import {
    setReleasesData,
    clearReleasesData,
    releasesDataReducer,
} from '../../ducks/newReleases';

const mockObject = [
    {
        mockVariable: 'mockValue',
    }
];

describe('newReleases Duck Tests', () => {
    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setReleasesData should match the snapshot' , () => {
            expect(setReleasesData(mockObject)).toMatchSnapshot();
        });
        it('clearReleasesData should match the snapshot' , () => {
            expect(clearReleasesData()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(releasesDataReducer()).toEqual([]);
        });
        it('should return the passed state if called with no action' , () => {
            expect(releasesDataReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(releasesDataReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should set the value when called with setReleasesData' , () => {
            expect(releasesDataReducer('', setReleasesData(mockObject))).toEqual(mockObject);
        });
        it('should clear the store when called with clearReleasesData' , () => {
            expect(releasesDataReducer(mockObject, clearReleasesData())).toEqual([]);
        });
    });
});