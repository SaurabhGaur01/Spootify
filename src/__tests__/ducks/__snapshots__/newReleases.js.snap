// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newReleases Duck Tests Action Generator clearReleasesData should match the snapshot 1`] = `
Object {
  "type": "modules/spootify-app/CLEAR_RELEASES_DATA",
}
`;

exports[`newReleases Duck Tests Action Generator setReleasesData should match the snapshot 1`] = `
Object {
  "data": Array [
    Object {
      "mockVariable": "mockValue",
    },
  ],
  "type": "modules/spootify-app/SET_RELEASES_DATA",
}
`;
